### PR TITLE
Adjust availability messaging based on the work by masteradhoc

### PR DESCRIPTION
**Pull request by borkweb**
These updates add a few filters and functions for spitting out messaging around ticket availability that observes multiple ticket types and availabilities.

Related: https://github.com/moderntribe/event-tickets-plus/pull/139

See: https://central.tri.be/issues/45467

### CODE REVIEW

**Comment by tr1b0t** [comment]
Status|Message|Line
:-----:|-----|-----
:x: | Don't leave debug_robot lying around | [RSVP.php:175](https://github.com/moderntribe/event-tickets/blob/bbcdded651814634d823b4995604a3c85f503a99/src/Tribe/RSVP.php#L175)
:x: | TRUE, FALSE and NULL must be lowercase; expected &quot;true&quot; but found &quot;TRUE&quot; | [RSVP.php:175](https://github.com/moderntribe/event-tickets/blob/bbcdded651814634d823b4995604a3c85f503a99/src/Tribe/RSVP.php#L175)

> via jenkins/codesniffer-phpcs

File: src/Tribe/Ticket_Object.php [hunk 1]
```diff
@@ -210,38 +220,97 @@ public function date_in_range( $datetime ) {
...
+
+			if ( $this->date_in_range( $timestamp ) ) {
+				$slug = 'available';
+			} elseif ( $this->date_is_earlier( $timestamp ) ) {
+				$slug = 'availability-future';
+			} elseif ( $this->date_is_later( $timestamp ) ) {
+				$slug = 'availability-past';
+			}
+
+			return $slug;
```
Comment (barryhughes) [inline src/Tribe/Ticket_Object.php]: Worth making it filterable?

Comment (borkweb) [inline src/Tribe/Ticket_Object.php]: Yeah! Added :)

**Comment by barryhughes** [comment]
:o:

**Pull request merged**